### PR TITLE
8268641: [foreign] assert(allocates2(pc)) failed: not in CodeBuffer memory with ShenandoahGC

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

The following 4 tests failed with -XX:+UseShenandoahGC on x86 (both release and debug VMs).
```
java/foreign/stackwalk/TestStackWalk.java
java/foreign/valist/VaListTest.java
java/foreign/TestUpcall.java
java/foreign/StdLibTest.java
```

The reason is that code buffer size for upcall_stub_linkToNative is too small.
To fix the crash, the size has been increased from 1024 to 2048 (note: 1536 is still not enough).

Thanks.
Best regards,
Jie

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8268641](https://bugs.openjdk.java.net/browse/JDK-8268641): [foreign] assert(allocates2(pc)) failed: not in CodeBuffer memory with ShenandoahGC

### Reviewers
 * [Rickard Bäckman](https://openjdk.java.net/census#rbackman) (@rickard - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/35/head:pull/35` \
`$ git checkout pull/35`

Update a local copy of the PR: \
`$ git checkout pull/35` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/35/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 35`

View PR using the GUI difftool: \
`$ git pr show -t 35`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/35.diff">https://git.openjdk.java.net/jdk17/pull/35.diff</a>

</details>

### CODE REVIEW

**Comment by DamonFool** [comment]
Could someone help to review this one-line fix?
Thanks.

**Review by rickard — APPROVED** [review]

**Comment by DamonFool** [comment]
Thanks @rickard for your review.

May I get a second review to finish this issue?
Thanks.

**Review by vnkozlov — APPROVED** [review]

**Comment by DamonFool** [comment]
Thanks @vnkozlov .
/integrate

**Pull request closed without merging**
